pkg/cachedinstances/factory.go

```diff
@@ -0,0 +1,85 @@
+package cachedinstances
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/atlassian/gostatsd"
+	"github.com/atlassian/gostatsd/pkg/cachedinstances/cloudprovider"
+	"github.com/atlassian/gostatsd/pkg/cloudproviders/aws"
+	"github.com/atlassian/gostatsd/pkg/cloudproviders/k8s"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"golang.org/x/time/rate"
+)
+
+var (
```

Comment:
Two types of defaults will be gone once k8s provider is no longer a provider but a direct implementation of `CachedInstances` interface (next PR).